### DISCUSSION TITLE
📢 New Feature: DNS Service

### DISCUSSION THREAD

**Discussion by eduardogsilva** [eduardogsilva/wireguard_webadmin#22] [Announcements]
I'm excited to announce that I've just integrated a DNS service into `wireguard_webadmin`. This feature enables the use of static DNS entries which are only resolvable within your VPN (e.g., `serverB.local` resolving to a private IP). Here’s how you can take advantage of this new feature:

- Navigate to the DNS tab in the interface and add your static DNS entries.
- Configure your WireGuard instance to serve its own private VPN IP as a DNS server.
- If you don't use static DNS entries, you can still use the DNS service for improved privacy.

**Important Update Requirements:**
To upgrade to this version, you'll need to update your Docker Compose file. Additionally, manual updates to your peer's configurations might be necessary to utilize this new DNS service effectively.

This enhancement aims to further streamline network management and security for all `wireguard_webadmin` users.

Feel free to share your feedback and suggestions on these updates!

**Comment by xgusto** [comment]
I don't know if I understand it correctly. I have wg network 10.10.3.0/24.

`DNS --> DNS settings --> Primary Resolver* `
`10.10.3.1`

`DNS --> Add static Host`

```
Hostname*: myrouter.ab
Ip address*: 10.10.3.3
save
````
`Restart wireguard service`

I connect to my router via 10.10.3.3 and everything works fine. When I use the hostname myrouter.ab, it doesn't work.
Yesterday I updated docker to Version 0.9611

**Comment by eduardogsilva** [comment]
Hello @xgusto .

Can you please check if the DNS iptables rules are in place?

Just check the Console > Firewall 

Grab a screenshot from the WGWADM_PREROUTING chain.
Just like this:

![image](https://github.com/eduardogsilva/wireguard_webadmin/assets/8627839/e62679f9-8896-4470-8cb3-22b2fddfd72d)



Cheers!

**Comment by xgusto** [comment]
I have this there
```
Chain WGWADM_PREROUTING (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 DNAT       17   --  wg0    *       0.0.0.0/0            10.10.3.1            udp dpt:53 to:127.0.0.250:53
    0     0 DNAT       6    --  wg0    *       0.0.0.0/0            10.10.3.1            tcp dpt:53 to:127.0.0.250:53
    0     0 DNAT       17   --  wg1    *       0.0.0.0/0            10.10.4.1            udp dpt:53 to:127.0.0.250:53
    0     0 DNAT       6    --  wg1    *       0.0.0.0/0            10.10.4.1            tcp dpt:53 to:127.0.0.250:53
```

**Reply by eduardogsilva** [reply]
the 127.0.0.250 address is a fallback if the resolution to the dns container name fails.

Check if the dns container is running. 

please, also check if the resolution is working properly.

```
docker exec -it wireguard-webadmin bash
root@ce2100edb178:/app# ping wireguard-webadmin-dns
```

Paste the results in a new discussion, I will be locking this topic since its a feature announcement. 

Cheers.
